checks/raw/binary_artifact.go

```diff
@@ -54,7 +54,7 @@ func BinaryArtifacts(c clients.RepoClient) (checker.BinaryArtifactData, error) {
 	err := fileparser.OnMatchingFileContentDo(c, fileparser.PathMatcher{
```

Comment:
I probably would use a second function that calls this one rather than change all the callers.

Reply:
Agreed, that's what I had intended with https://github.com/ossf/scorecard/pull/3473#discussion_r1323343891

Reply:
The problem is the function does not do what its documentation and name suggest. The documentation says: ```// OnMatchingFileContentDo matches all files ... and on every successful match, runs onFileContent fn on the file's contents.```. However, the function does not match all files as there is filtering going on, which at the moment isn't clear until reading the actual loop that does the filtering -- and it was confusing to debug as someone new to the code.

Adding the argument and changing the callers is an effort to correct this.

I saw the options as:
1) Change the name of the function + change the documentation `OnMatchingFileContentDo` --> `OnMatchingNonTestFileContentDo` keep the argument list.
2) Add arguments to the function to make it more obvious that filtering is happening and make it possible to avoid the filtering
3) Add a new function that allows for actually matching all files and keep the semantics of `OnMatchingFileContentDo` as is (or perhaps minor refactoring to avoid duplicate logic), but renaming it to indicate the matching happens on a subset of files.

I went for 2, but all of the above options require changes to all the callers.

Reply:
I would say there's an option 4.

Add a new function, leave the existing function as is (perhaps clarifying the comment) and adding a new function called `OnMatchingFileContentWithTestDataDo` (or similar). Which from a backwards compatibility standpoint is the best.

Reply:
I'll address this in a follow-up PR. Left out changing the logic of the file matching for now.